### DISCUSSION TITLE
Smart Caps Lock remapping doesn't take effect when .xprofile is sourced

### DISCUSSION THREAD

**Discussion by josh5555s** [regolith-linux/regolith-desktop#676] [Q&A] [answered]
I'm on Regolith 2.1 Linux. I can run ~/.xprofile and it works, but it doesn't run on startup.

.xprofile permissions are -rwxrwxrwx

File contents are: 
\#make CapsLock behave like Ctrl: 
setxkbmap -option ctrl:nocaps 
\#make short-pressed Ctrl behave like Escape: 
xcape -e 'Control_L=Escape' 

I'd love for smart caps lock to be enabled automatically. Thanks for any help.

**Comment by kgilmer** [comment, marked as answer]
Hi @josh5555s , did you get Regolith via packages or did you install the ISO?  It seems that the display manager should be sourcing `.xprofile` according to [these docs](https://wiki.archlinux.org/title/xprofile) and I did verify that `gdm3` does in fact seem to test and source the file.

**Reply by josh5555s** [reply]
The ISO. You are right, when I add` touch ~/testing.txt` that does work, so it is the setxkbmap/xcape commands that aren't working when .xprofile is sourced originally. But then they do work when it is sourced again manually.

**Reply by josh5555s** [reply]
The issue remains, although my original understanding of the cause was wrong. The Smart Caps Lock commands, [taken from here](https://gist.github.com/tanyuan/55bca522bf50363ae4573d4bdcf06e2e), have always worked for me in other ubuntu distros. There is something about the way that Regolith starts up that is getting in the way. Any ideas would be much appreciated. Thanks.

**Reply by josh5555s** [reply]
I finally found a solution here: [https://unix.stackexchange.com/questions/66624/where-is-xkb-getting-its-configuration/66657](https://unix.stackexchange.com/questions/66624/where-is-xkb-getting-its-configuration/66657)

The issue is that gnome overwrites the xkb configs, and so I had to use gsettings to directly update xkb-options. 

Lastly, I still needed `xcape -e 'Control_L=Escape'` to run, but for whatever reason it doesn't run from .xprofile. However, I found that putting it in a bash script, then calling the bash script from .xprofile does work. I don't know why that's necessary, but it works!

I hope this helps anyone else who wants to use [Smart Caps Lock](https://gist.github.com/tanyuan/55bca522bf50363ae4573d4bdcf06e2e).
